experimental/packages/otlp-exporter-base/src/transport/fetch-transport.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { IExporterTransport } from '../exporter-transport';
+import { ExportResponse } from '../export-response';
+import { diag } from '@opentelemetry/api';
+import {
+  isExportRetryable,
+  parseRetryAfterToMills,
+} from '../is-export-retryable';
+
+export interface FetchTransportParameters {
+  url: string;
+  headers: () => Record<string, string>;
+}
+
+class FetchTransport implements IExporterTransport {
+  constructor(private _parameters: FetchTransportParameters) {}
+
+  async send(data: Uint8Array, timeoutMillis: number): Promise<ExportResponse> {
+    let timeout: ReturnType<typeof setTimeout> | undefined;
+    try {
+      const abortController = new AbortController();
+      timeout = timeoutMillis
+        ? setTimeout(() => abortController.abort(), timeoutMillis)
+        : undefined;
+
+      const base = globalThis.location?.href;
+      // NOTE: In non-browser environments, `globalThis.location` will be `undefined`.
+      const isBrowserEnvironment = !!base;
+      const url = new URL(this._parameters.url, base);
+      const response = await fetch(url.href, {
+        method: 'POST',
+        headers: this._parameters.headers(),
+        body: data,
+        signal: abortController.signal,
+        mode: isBrowserEnvironment
```

Comment:
For a browser environment, `keepalive: true` is highly desirable. So a fetch request won't be canceled when a user closes a tab.

`navigator.sendBeacon` also doesn't cancel a request a when user closes tab.

Reply:
I agree, would also like to have this set right from the get-go.